engines/python/setup/djl_python/deepspeed.py

```diff
@@ -10,98 +10,246 @@
+    "text-classification": AutoModelForSequenceClassification,
+    "question-answering": AutoModelForQuestionAnswering,
+    "fill-mask": AutoModelForMaskedLM,
+    "token-classification": AutoModelForTokenClassification,
+    "conversational": AutoModelForCausalLM,
+}
+
+MODEL_TYPE_TO_INJECTION_POLICY = {
+    "roberta": {transformers.models.roberta.modeling_roberta.RobertaLayer: HFBertLayerPolicy},
+    "gpt2": {transformers.models.gpt2.modeling_gpt2.GPT2Block: HFGPT2LayerPolicy},
+    "bert": {transformers.models.bert.modeling_bert.BertLayer: HFBertLayerPolicy},
+    "gpt_neo": {transformers.models.gpt_neo.modeling_gpt_neo.GPTNeoBlock: HFGPTNEOLayerPolicy},
+    "gptj": {transformers.models.gptj.modeling_gptj.GPTJBlock: HFGPTJLayerPolicy},
+    "opt": {transformers.models.opt.modeling_opt.OPTDecoderLayer: HFOPTLayerPolicy},
+    "gpt-neox": {transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXLayer: GPTNEOXLayerPolicy},
+    "bloom": {transformers.models.bloom.modeling_bloom.BloomBlock: BLOOMLayerPolicy},
+}
 
 
 class DeepSpeedService(object):
 
     def __init__(self):
-        self.predictor = None
-        self.max_new_tokens = 0
+        self.pipeline = None
         self.initialized = False
+        self.ds_config = None
+        self.task = None
+        self.logger = logging.getLogger()
+        self.model_dir = None
+        self.model_id = None
+        self.data_type = None
+        self.max_tokens = None
+        self.device = None
+        self.world_size = None
+        self.tensor_parallel_degree = None
+        self.model_config = None
 
     def initialize(self, properties: dict):
-        self.max_new_tokens = int(properties.get("max_new_tokens", "50"))
-        model_dir = properties.get("model_dir")
-        data_type = properties.get("data_type", "fp32")
-        mp_size = int(properties.get("tensor_parallel_degree", "1"))
-        model_id = properties.get("model_id")
-        # LOCAL_RANK env is initialized after constructor
-        device = int(os.getenv('LOCAL_RANK', '0'))
-        if not model_id:
-            model_id = model_dir
-            config_file = os.path.join(model_dir, "config.json")
+        self.parse_properties(properties)
+        self.validate_model_type_and_task()
+        self.create_model_pipeline()
+        self.logger.info(f"Initialized DeepSpeed model with the following configurations"
+                         f"model: {self.model_id}"
+                         f"task: {self.task}"
+                         f"data_type: {self.data_type}"
+                         f"tensor_parallel_degree: {self.tensor_parallel_degree}")
+        self.initialized = True
+
+    def parse_properties(self, properties):
+        self.model_dir = properties.get("model_dir")
+        self.model_id = properties.get("model_id")
+        self.task = properties.get("task")
+        self.data_type = properties.get("data_type", "fp32")
+        self.max_tokens = int(properties.get("max_tokens", 1024))
```

Comment:
what should be a proper default to max_tokens?

Reply:
Deepspeed uses 1024 as the default, so I think keeping it the same as that makes sense. 

Though, i think task specific pipelines on HF side might override that and set other defaults.